.gitlab-ci.yml

```diff
@@ -301,6 +308,114 @@ dogstatsd_suse-x64:
+    - echo "$APT_SIGNING_KEY_PASSPHRASE" | deb-s3 upload -c $DEB_RPM_TESTING_BUCKET_BRANCH -b $DEB_TESTING_S3_BUCKET -a x86_64 --sign=$APT_SIGNING_KEY_ID --gpg_options="--passphrase-fd 0 --pinentry-mode loopback --batch --digest-algo SHA512" --preserve_versions --visibility public $OMNIBUS_PACKAGE_DIR/*amd64.deb
+
+# deploy rpm packages to yum staging repo
+deploy_rpm_testing:
+  stage: deploy_testing
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent-builders/deploy:latest
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR
+  only:
+    - master
+    - tags
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - source /usr/local/rvm/scripts/rvm
+    - rvm use 2.4
+    - mkdir -p ./rpmrepo/x86_64/
+    - aws s3 sync s3://$RPM_TESTING_S3_BUCKET/$DEB_RPM_TESTING_BUCKET_BRANCH/ ./rpmrepo/
+    - cp $OMNIBUS_PACKAGE_DIR/*x86_64.rpm ./rpmrepo/x86_64/
+    - createrepo --update -v --checksum sha ./rpmrepo/x86_64
+    - aws s3 sync ./rpmrepo/ s3://$RPM_TESTING_S3_BUCKET/$DEB_RPM_TESTING_BUCKET_BRANCH/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
+
+# deploy rpm packages to yum staging repo
+deploy_suse_rpm_testing:
+  stage: deploy_testing
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent-builders/deploy:latest
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR_SUSE
+  only:
+    - master
+    - tags
+    - greg/testkitchen
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - source /usr/local/rvm/scripts/rvm
+    - rvm use 2.4
+    - mkdir -p ./rpmrepo/suse/x86_64/
+    - aws s3 sync s3://$RPM_TESTING_S3_BUCKET/suse/$DEB_RPM_TESTING_BUCKET_BRANCH/ ./rpmrepo/
+    - cp $OMNIBUS_PACKAGE_DIR_SUSE/*x86_64.rpm ./rpmrepo/x86_64/
+    - createrepo --update -v --checksum sha ./rpmrepo/x86_64
+    - aws s3 sync ./rpmrepo/ s3://$RPM_TESTING_S3_BUCKET/suse/$DEB_RPM_TESTING_BUCKET_BRANCH/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
+
+# run dd-agent-testing
+dd_agent_testing:
+  stage: dd_agent_testing
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent-builders/dd-agent-testing:latest
+  only:
+    - master
+    - greg/testkitchen
+    - tags
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - cd $DD_AGENT_TESTING_DIR
+    - cp .kitchen-azure.yml .kitchen.yml
+    - bash -l tasks/testall.sh
+  artifacts:
+    expire_in: 2 weeks
+    paths:
+      - test/kitchen/.kitchen/logs
+
+# run dd-agent-testing
+cleanup_testing:
+  stage: cleanup_testing
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent-builders/deploy:latest
+  only:
+    - master
+    - greg/testkitchen
+    - tags
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - aws s3 rm --recursive s3://DEB_TESTING_S3_BUCKET/pool
```

Comment:
I don't see how that cannot fail if two pipelines run in parallel. It's the same for the `deploy` and `test` stages, but even more obvious here.
We need to use `$CI_PIPELINE_ID` to isolate every build pipeline in its own prefix to avoid interferences. Can't we run the debian repo in `s3://PACKAGE_TEST_S3_BUCKET/$CI_PIPELINE_ID/deb` and rpms in `s3://PACKAGE_TEST_S3_BUCKET/$CI_PIPELINE_ID/rpm` so the cleanup step can just remove `s3://PACKAGE_TEST_S3_BUCKET/$CI_PIPELINE_ID/` and be done with it?

Reply:
You're right -- I will need to figure out how to do this